cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/DatabaseImpl.java

```diff
@@ -928,7 +890,7 @@ public Void call(SQLDatabase db) throws Exception {
                             if (newWinnerTx.getClass().equals(DocumentRevision.class)) {
                                 // user gave us a new DocumentRevision instance
                                 isNewOrUpdatedRevision = true;
-                            } else if (newWinnerTx.getClass().equals(InternalDocumentRevision.class)) {
+                            } else if (newWinnerTx instanceof InternalDocumentRevision) {
```

Comment:
is instanceof correct here? we need to distinguish between subtypes and supertypes and I thought that getClass() was the better check for this